### DISCUSSION TITLE
Networked Gameplay Design

### DISCUSSION THREAD

**Discussion by capalmer1013** [capalmer1013/2d-survival#33] [Ideas]
what are good/important things to consider with networking design?

**Comment by capalmer1013** [comment]
@Jarvl since networked objects don't need any logic tied to them I think the minimal serialized information could be something like this.
U, V (sprite map coordinates for what to draw)
h, w (height and width of object on sprite map
x, y (where on the world map to draw the sprite)

**Reply by capalmer1013** [reply]
so it could just be a list of 6-tuples

**Reply by Jarvl** [reply]
I think it could be less than that. Might be best to dispatch more specific events like `OBJECT_NEW_POSITION`, `OBJECT_REMOVED`, `OBJECT_ADDED` instead of a general purpose `OBJECT_UPDATED` event. That way we only need to serialize data specific to the type of event happening, instead of all the object data.

**Reply by capalmer1013** [reply]
yes
